packages/terminal/src/browser/terminal-frontend-module.ts

```diff
@@ -66,6 +70,22 @@ export default new ContainerModule(bind => {
         }
     }));
 
+    bind(TerminalSearchWidgetFactory).toDynamicValue(ctx => (terminal: Terminal, node: Element, terminalWdgId: string) => {
```

Comment:
please reimplement it and apply the approach from tree widgets: https://github.com/theia-ide/theia/blob/c593575a1f143dfa0ecd22b32adf34b840ec0fa2/packages/navigator/src/browser/navigator-frontend-module.ts#L44-L46